### PR TITLE
Rewrite and reorganise README to help readers evaluate extension

### PR DESCRIPTION
**Pull request by itowlson**
The README is in a bit of a muddled state, much of it dating back to the handover from its parents.  We want to lead on information that will help potential users to evaluate whether it is going to be worth their while.

With this change the README leads on the target audience and the feature list, and refines the prerequisites into a 'getting started' section (moving the "except if the moon is in Pisces" stuff out to footnotes).

### CODE REVIEW

**Comment by itowlson** [comment]
Fixes #384.

File: README.md [hunk 1]
```diff
@@ -1,69 +1,74 @@
...
+  pods and nodes.
+* Browse Helm repos and install charts into your Kubernetes cluster.
+* Intellisense for Kubernetes resources and Helm charts and templates.
+* Edit Kubernetes resource manifests and apply them to your cluster.
+* Build and run Dockerfiles in your cluster.
+* View diffs of a resource's current state against the resource manifest in your
+  Git repo
+* Easily check out the Git commit corresponding to a deployed application.
+* Run commands or start a shell within your application's pods.
+* Get logs and events from your clusters.
```
Comment (bnookala) [inline README.md]: "Get and follow logs or events from your cluster"

File: README.md [hunk 2]
```diff
@@ -1,69 +1,74 @@
...
-
-For setting up `draft` you can provide a path to the binary via configuration (`vs-kubernetes.draft-path`) if it is not on your PATH.
-
-### Setting up the image repository path
+An extension for developers building applications to run in Kubernetes clusters,
+and for DevOps staff troubleshooting Kubernetes applications.  Features include:
+
+* View your clusters in an explorer tree view, and drill into workloads, services,
+  pods and nodes.
+* Browse Helm repos and install charts into your Kubernetes cluster.
```
Comment (bnookala) [inline README.md]: Perhaps logically group Helm/Draft features together (move this to be with its bullet-y friends at the bottom)

File: README.md [hunk 3]
```diff
@@ -1,69 +1,74 @@
...
+* Edit Kubernetes resource manifests and apply them to your cluster.
+* Build and run Dockerfiles in your cluster.
+* View diffs of a resource's current state against the resource manifest in your
+  Git repo
+* Easily check out the Git commit corresponding to a deployed application.
+* Run commands or start a shell within your application's pods.
+* Get logs and events from your clusters.
+* Create Helm charts using scaffolding and snippets.
+* Bootstrap applications using Draft, and rapidly deploy and debug them to speed up
+  the development loop.
```
Comment (bnookala) [inline README.md]: Also worth a mention: Raise a shell in a running container, port forward to a running container

File: README.md [hunk 4]
```diff
@@ -1,69 +1,74 @@
...
+
+**What's new in this version?**  See the [change log](CHANGELOG.md) to find out!
+
+## Getting started with the extension
+
+### Dependencies
+
+The Kubernetes extension may need to invoke the following command line tools, depending on
+which features you use.  You will probably need `kubectl` and `docker` at minimum.
```
Comment (bnookala) [inline README.md]: I feel like even docker is un-necessary if you're strictly only using the extension to browse around and view resources

File: README.md (hunk 3 above)
Comment (bnookala) [reply README.md]: Oh derp - didn't end up seeing the second part of that sentence. Carry on!

File: README.md [hunk 5]
```diff
@@ -1,69 +1,76 @@
...
-### Setting up the image repository path
+An extension for developers building applications to run in Kubernetes clusters,
+and for DevOps staff troubleshooting Kubernetes applications.  Features include:
+
+* View your clusters in an explorer tree view, and drill into workloads, services,
+  pods and nodes.
+* Browse Helm repos and install charts into your Kubernetes cluster.
+* Intellisense for Kubernetes resources and Helm charts and templates.
+* Edit Kubernetes resource manifests and apply them to your cluster.
+* Build and run Dockerfiles in your cluster.
```
Comment (radu-matei) [inline README.md]: nit: maybe containers / container images might be a better wording for this?
(You don't necessarily run a Dockerfile)

**Pull request merged**